### PR TITLE
IPC::Open3/Open2: support dup mode for lexical filehandles

### PR DESCRIPTION
**Pull request by mauke**
I had to invent a new calling syntax/convention because the existing code relies on parsing an open mode and bareword filehandle name from a single string argument.

The documentation has been reworked to hopefully make it clearer in which cases IPC::Open3 uses arguments as outputs (to place filehandles into) and inputs (to read filehandles from).

---------------------------------------------------------------------------------
* This set of changes requires a perldelta entry, and it is included.

### CODE REVIEW

File: ext/IPC-Open3/lib/IPC/Open2.pm [hunk 1]
```diff
@@ -22,45 +22,96 @@ IPC::Open2 - open a process for both reading and writing using open2()
...
-    my $pid = open(my $fh, "|cmd args|");
+    my $pid = open(my $fh, "|cmd args|");  # ERROR
+
+but you have to write it as:
+
+    my $pid = open2($chld_out, $chld_in, @command_and_args);
+
+The C<$chld_in> filehandle will have autoflush turned on.
+
+By default, the filehanndles you pass in are used as output parameters.
```
Comment (tonycoz) [inline ext/IPC-Open3/lib/IPC/Open2.pm]: extra n in "filehanndles"

File: ext/IPC-Open3/lib/IPC/Open3.pm [hunk 2]
```diff
@@ -40,50 +40,101 @@ IPC::Open3 - open a process for reading, writing, and error handling using open3
...
+It connects C<$chld_in> for writing to the child's standard input, C<$chld_out>
+for reading from the child's standard output, and C<$chld_err> for reading from
+the child's standard error stream.  If C<$chld_err> is false, or the same file
+descriptor as C<$chld_out>, then C<STDOUT> and C<STDERR> of the child are on
+the same filehandle.  This means that you cannot pass an uninitialized variable
+for C<$chld_err> and have C<open3> auto-generate a filehandle for you, but
+gensym from L<Symbol> can be used to vivify a new glob reference; see
+L</SYNOPSIS>.  The C<$chld_in> handle will have autoflush turned on.
+
+By default, the filehanndles you pass in are used as output parameters.
```
Comment (tonycoz) [inline ext/IPC-Open3/lib/IPC/Open3.pm]: Extra n here too

**Comment by tonycoz** [comment]
Looks fine except for the typos

File: ext/IPC-Open3/lib/IPC/Open2.pm (hunk 1 above)
Comment (mauke) [reply ext/IPC-Open3/lib/IPC/Open2.pm]: Thanks, fixed.

**Review by tonycoz — APPROVED** [review]

**Pull request merged**